pandas/io/parsers/base_parser.py

```diff
@@ -138,31 +132,20 @@ def __init__(self, kwds) -> None:
 
             # validate index_col that only contains integers
             if self.index_col is not None:
-                is_sequence = isinstance(self.index_col, (list, tuple, np.ndarray))
                 if not (
-                    is_sequence
+                    is_list_like(self.index_col, allow_sets=False)
```

Comment:
aren't the same critera we all in header? 

can we change the `validate_header` routine to do this validate also (e.g. make it more generic) then can just call here as well

Reply:
There are at two differences:
1. `index_col` allows negative numbers while `header` does not. (I can't find any examples of negative index_col inputs in the documentation or the tests, but they work.)
2. `index_col` can contain ints in any order, while `header` breaks on non-increasing ints. There's a separate issue suggesting that we raise a ValueError to prevent non-increasing header arguments (#47011)

It would be nice to consolidate the `index_col` validation currently spread over readers.py and base_reader.py into a single function, but maybe it's cleaner to call it `validate_index` as first step and see how similar it is to `validate_header`. Could I look into that in a separate PR?